File: ois/ois20/round3/ois11_round3_solutions.tar(1)/police.pas
{ constraints }
const
    MAXN = 100000;

{ input data }
var
    N, K, i, pos, sol : longint;
    V        : array[0..MAXN] of longint;
    vis      : array[0..MAXN] of boolean;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N, K);
    for i:=1 to N do
    begin
        read(V[i]);
        vis[i] := false;
    end;
    readln();

    pos := 1;
    sol := 0;
    while (pos <> K) and (vis[pos] = false) do
    begin
        vis[pos] := true;
        pos := V[pos];
        sol := sol + 1;
    end;

    if pos = K
    then
        writeln(sol)
    else
        writeln(-1)
    end.
end.

File: ois/ois20/round3/ois11_round3_solutions.tar(1)/police.py
#!/usr/bin/env python3

# input data
N, K = map(int, input().strip().split())
V = [-1] + list(map(int, input().strip().split()))
vis = [False] * (N+1)

pos = 1
sol = 0
while pos != K and not vis[pos]:
    vis[pos] = True
    pos = V[pos]
    sol += 1

if pos == K:
    print(sol)
else:
    print(-1)
